### PR TITLE
statement-store: Remove double peer state clean up on flooding and add zombienet tests

### LINKED ISSUE #11535: statement-store: close resilience under load test coverage gaps
### Summary

Tracking issue for the remaining resilience under load scenarios in statement-store tests. See details in the [testing strategy](https://docs.google.com/document/d/1dMEkH82ZxFdUN_vX-Qhl2kdiHVVxD67AYhYCTmTIZl4/edit?tab=t.1d4x55fgv2og).

### Tasks

**P0**
- [ ] Statement flooding: Zombienet tests cover detection edge cases and rejection of oversized statements, verify malicious peers are rate-limited while legitimate traffic remains unaffected https://github.com/paritytech/polkadot-sdk/pull/11701
- [x] Throughput under expected load: Versi tests use the expected load model, verify latency stays within thresholds, and check for no memory leaks or latency creep over extended runs https://github.com/paritytech/parity-devnet/pull/240
- [x] Stress handling: Versi tests verify sudden stress load do not cause statement loss, statements reach all nodes, and subscriptions remain reliable. https://github.com/paritytech/parity-devnet/pull/240
- [x] Resource overhead: Versi tests compare resource usage against a baseline without statement-store and produce recommended hardware guidance https://github.com/paritytech/parity-devnet/pull/239

### PR DESCRIPTION
**Pull request by AndreiEres**
# Description

Closes https://github.com/paritytech/polkadot-sdk/issues/11535: covers flooding detection.

Fixed a bug with double peer state cleanup. The initial purpose was to add flooding e2e tests, but the tests revealed a bug in the rate-limit mechanics: the peer was attempted to be removed twice.

Added two zombienet integration tests that verify statement-store flooding detection:

- Sustained-rate flooding — submits statements continuously against a rate-limited node. Verifies the
node accepts early batches (token bucket not yet exhausted) then triggers flooding detection as tokens
drain.
- Burst flooding — submits a large batch at once against a tightly rate-limited node. Verifies immediate
 flooding detection with zero statements accepted.

Also refactors common test helpers to allow spawning custom network configurations for the flooding
tests.

## Integration

No integration needed.

### CODE REVIEW

File: cumulus/zombienet/zombienet-sdk/tests/zombie_ci/statement_store/common.rs [hunk 1]
```diff
@@ -180,23 +178,47 @@ fn create_chain_spec_with_allowances(
...
+	.iter()
+	.map(|s| s.as_str().into())
+	.collect()
+}
+
+pub(super) fn base_dir() -> Result<PathBuf, anyhow::Error> {
+	let path = std::env::var("ZOMBIENET_SDK_BASE_DIR")
+		.ok()
+		.map(PathBuf::from)
+		.unwrap_or_else(|| std::env::temp_dir().join(format!("zombienet-{}", std::process::id())));
```
Comment (pepoviola) [inline cumulus/zombienet/zombienet-sdk/tests/zombie_ci/statement_store/common.rs]:
```suggestion
		.unwrap_or_else(|| std::env::temp_dir().join(format!("zombie-{}", std::process::id())));
```
we need to sue `zombie-<random>` to pick the logs.

File: cumulus/zombienet/zombienet-sdk/tests/zombie_ci/statement_store/integration.rs [hunk 2]
```diff
@@ -205,6 +205,188 @@ async fn statement_store_check_propagation_and_quota_invariants() -> Result<(),
...
+		},
+		10u64,
+	)
+	.await?;
+	assert!(
+		bob_submitted.get() > 0.0,
+		"Bob should have accepted early batches before flooding (got {})",
+		bob_submitted.get()
+	);
+	info!("Bob accepted {} statements before flooding (sustained, not burst)", bob_submitted.get());
```
Comment (alexggh) [inline cumulus/zombienet/zombienet-sdk/tests/zombie_ci/statement_store/integration.rs:346]: Can we also check the disconnect happens ?

**Review by alexggh — APPROVED** [review]

File: cumulus/zombienet/zombienet-sdk/tests/zombie_ci/statement_store/integration.rs (hunk 2 above)
Comment (AndreiEres) [reply cumulus/zombienet/zombienet-sdk/tests/zombie_ci/statement_store/integration.rs:346]: Done

**Comment by P1sar** [comment]
I have test   `test_peer_disconnected_on_flooding` falling localy bcs it checks that peers are cleared fron known peers
https://github.com/paritytech/polkadot-sdk/blob/a9ffa03ac6738da49622d81a64a1bb4809911464/substrate/client/network/statement/src/lib.rs#L2486

I have a feeling that this is a correct behaviour generaly so we can't jsut remove this check

**Comment by AndreiEres** [comment]
> I have test `test_peer_disconnected_on_flooding` falling localy bcs it checks that peers are cleared fron known peers

Good catch


> I have a feeling that this is a correct behaviour generaly so we can't jsut remove this check

You're right. We can emulate the event and check that in the end it gets cleared.

**Review by P1sar — APPROVED** [review]
